mysql/datadog_checks/mysql/statements.py

```diff
@@ -221,12 +221,16 @@ def _query_summary_per_statement(self):
         several fields must be further processed from the delta values.
         """
         only_query_recent_statements = self._config.statement_metrics_config.get('only_query_recent_statements', False)
+        # Since `performance_schema.prepared_statements_instances` doesn't have a last_seen column,
+        # we only collect prepared statements if we're not querying recent statements
+        collect_prepared_statements = not only_query_recent_statements and self._config.statement_metrics_config.get(
```

Comment:
Is there a reason we're squeezing this into this existing query given the nuance? This is confusing for a customer to have to reason. This is also technically a breaking change in functionality as customers who explicitly set `only_query_recent_statements=true` will have this now disabled on upgrade. We should think through how we can accomplish this without introducing a breaking change

Reply:
Apologies, i misread this myself and take back my comment about breaking change. But still this is a confusing. Could we just issue multiple queries?